### PR TITLE
Added multi-port service support for F5 loadbalancer

### PR DESCRIPTION
**Pull request by abithap**

### CODE REVIEW

File: loadbalancer/loadbalancer/backend/backends/f5/loadbalancer.go [hunk 1]
```diff
@@ -106,112 +106,137 @@ func (ctr *F5Controller) HandleConfigMapCreate(configMap *api.ConfigMap) error {
...
+		if p.NodePort == 0 {
+			err = fmt.Errorf("NodePort is needed for loadbalancer")
+			return err
+		}
+
+		monitorName := utils.GetResourceName(MONITOR, name, servicePortName)
+		monExist, err := ctr.monitorExist(monitorName)
 		if err != nil {
-			err = fmt.Errorf("Could not create monitor %v. %v", monitorName, err)
+			err = fmt.Errorf("Error accessing monitors. %v", err)
```
Comment (kokhang) [inline loadbalancer/loadbalancer/backend/backends/f5/loadbalancer.go]: Why did the error message changed? Looks like this applies to one monitor

File: loadbalancer/loadbalancer/backend/backends/f5/loadbalancer.go [hunk 2]
```diff
@@ -106,112 +106,137 @@ func (ctr *F5Controller) HandleConfigMapCreate(configMap *api.ConfigMap) error {
...
-		err = ctr.f5.CreateMonitor(monitorName, MONITOR_PROTOCOL, 5, 16, "", "")
+
+	for _, p := range serviceObj.Spec.Ports {
+		servicePortName := p.Name
+		if p.NodePort == 0 {
+			err = fmt.Errorf("NodePort is needed for loadbalancer")
+			return err
+		}
+
+		monitorName := utils.GetResourceName(MONITOR, name, servicePortName)
```
Comment (kokhang) [inline loadbalancer/loadbalancer/backend/backends/f5/loadbalancer.go]: You only need to create one monitor. No need to create multiple of them

File: loadbalancer/loadbalancer/utils/utils.go [hunk 3]
```diff
@@ -141,9 +141,9 @@ func NodeReady(node api.Node) bool {
 	return false
 }
 
-// GetLBConfigMapNodePortMap fetches all the configmaps and returns a map of loadbalancer configmaps to node port
-func GetLBConfigMapNodePortMap(client *unversioned.Client, configMapNamespace string, configMapLabelKey, configMapLabelValue string) map[string][]int {
-	configMapNodePortMap := make(map[string][]int)
+// GetLBConfigMapNodePortMap fetches all the configmaps and returns a map of loadbalancer pool to node port
+func GetLBConfigMapNodePortMap(client *unversioned.Client, configMapNamespace string, configMapLabelKey, configMapLabelValue string) map[string]int {
```
Comment (kokhang) [inline loadbalancer/loadbalancer/utils/utils.go]: Can you change the name of the method to reflect this change? Call it GetPoolNodePortMap or something like that

File: loadbalancer/loadbalancer/controllers/ipmanager.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ var ipConfigMapMutex sync.Mutex
 var empty struct{}
 
 type IPManager struct {
-	configMapName string
```
Comment (kokhang) [inline loadbalancer/loadbalancer/controllers/ipmanager.go]: Any reason why you changed the visibility of this variable?

Comment (abithap) [inline loadbalancer/loadbalancer/controllers/ipmanager.go]: yeah, ipManager configmap name is needed in the getbindIP method of f5

File: loadbalancer/loadbalancer/utils/utils.go (hunk 3 above)
Comment (abithap) [inline loadbalancer/loadbalancer/utils/utils.go]: sure. will change it in the next patch

File: loadbalancer/loadbalancer/backend/backends/f5/loadbalancer.go [hunk 5]
```diff
@@ -106,44 +106,45 @@ func (ctr *F5Controller) HandleConfigMapCreate(configMap *api.ConfigMap) error {
...
-				return err
-			}
-			glog.Infof("Monitor %v created.", monitorName)
-		}
-
 		poolName := utils.GetResourceName(POOL, name, servicePortName)
 		pool, err := ctr.f5.GetPool(poolName)
 		if err != nil {
 			err = fmt.Errorf("Error getting pool %v. %v", poolName, err)
-			defer ctr.deleteF5Resource(monitorName, MONITOR)
```
Comment (kokhang) [inline loadbalancer/loadbalancer/backend/backends/f5/loadbalancer.go]: why are we not deleting a monitor when an error happens? We should delete it

Comment (abithap) [inline loadbalancer/loadbalancer/backend/backends/f5/loadbalancer.go]: its deleted in the deletePreviouslyCreatedF5Resources() method

**Pull request merged**